tools/ci_build/github/azure-pipelines/mac-ios-ci-pipeline.yml

```diff
@@ -58,3 +63,34 @@ jobs:
           --path_to_protoc_exe $(Build.BinariesDirectory)/protobuf_install/bin/protoc \
           --parallel
       displayName: (CoreML EP) Build onnxruntime for iOS x86_64 and run tests using simulator
+
+- job: iOS_CI_XNNPACK_on_Mac
+  pool:
+    vmImage: 'macOS-11'
+  variables:
+    MACOSX_DEPLOYMENT_TARGET: '10.14'
+  timeoutInMinutes: 100
+  condition: or(eq('${{parameters.BuildXnnpack}}', 'true'), eq(variables['Build.SourceBranch'], 'refs/heads/main'))
```

Comment:
use template instead. 

Like:

```
      ${{ if eq(parameters.BuildXnnpack, 'true') }}:
```

Such statements are evaluated earlier than what you have. "condition" is evaluated at runtime.

Reply:
Like this one: https://github.com/microsoft/onnxruntime/blob/main/tools/ci_build/github/azure-pipelines/templates/py-packaging-stage.yml#L53

Reply:
I see, thanks

Reply:
Sorry, I probably gave you a wrong suggestion.  How about this: you move this job to a template file, then include that template file from here. Like
```yaml
  - ${{ if eq(parameters.enable_ubuntu_cpu, true) }}:
      - template: py-linux-ubuntu.yml
        parameters:
          arch: 'x86_64'
```

Then you create another pipeline that will be auto-triggered by every commit in the main branch, but not PRs. Then that pipeline's yaml file include this template too.  In this way, your mac-iso-ci-pipeline could still choose to run both jobs or just one of them.

Reply:
If you don't want to do that, your previous commit is fine too.

Reply:
If I want to control it with both triggerd by mannuly and merged into main, I have  to evaluated in the runtime. Because I can only get the parameters.BuildXnnpack when I set it as `true`, Is that right?

Reply:
No, it is not right. parameters are known when you submitted the build run. But at that time, most variables are still unknown. "runtime" means something is only known when the build step starts to run.

Reply:
Got it.